modules/ocf_stats/manifests/prometheus.pp

```diff
@@ -1,8 +1,25 @@
 # prometheus daemon config
 class ocf_stats::prometheus {
+  file {
+    '/usr/local/bin/gen-prometheus-nodes':
+      source => 'puppet:///modules/ocf_stats/prometheus/gen-prometheus-nodes',
+      mode   => '0755';
+  }
+
+  cron { 'gen-prometheus-nodes':
+    command => '/usr/local/bin/gen-prometheus-nodes /var/local/prometheus-nodes.json',
+    minute  => '03',
+    require => File['/usr/local/bin/gen-prometheus-nodes'];
+  }
+
+  exec { 'gen-promethues-nodes-initial':
+    command => 'usr/local/bin/gen-prometheus-nodes /var/local/prometheus-nodes.json',
+    onlyif  => 'test ! -f /var/local/prometheus-nodes.json',
```

Comment:
A better way to do this is to use the [`creates` parameter](https://puppet.com/docs/puppet/5.3/types/exec.html#exec-attribute-creates) here. I suppose my feelings are that having this exec here seems a bit strange since there's a cronjob to do the same thing but regularly, but I see the problem that this file won't exist for up to an hour on bootstrapping. If the file doesn't exist for that hour, does Prometheus error or does it just show no data?

Reply:
Haven't tested it, but I would expect Prometheus to just do nothing if the file doesn't exist, and then detect when the file gets created. I decided to put the exec command here since I think if we ever re-provision the stats machine, it would be nice to collect stats right away.